### PR TITLE
JazzyGrid: add theme

### PR DESCRIPTION
**Pull request by henriqueiamarino**
Your new link-in-bio site, JazzyGrid, organizes blocks to display contact information and links on an exciting landing page.
[Demo site](https://jazzygrid.mystagingwebsite.com/)

screenshot

### CODE REVIEW

File: jazzygrid/patterns/header.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+<div class="wp-block-group"><!-- wp:paragraph {"fontSize":"small"} -->
+<p class="has-small-font-size"><a href="#" target="_blank" rel="noreferrer noopener">Website</a></p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph {"fontSize":"small"} -->
+<p class="has-small-font-size"><a href="#" target="_blank" rel="noreferrer noopener">Albuns</a></p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph {"fontSize":"small"} -->
+<p class="has-small-font-size"><a href="#" target="_blank" rel="noreferrer noopener">Tickets</a></p>
```
Comment (ajlende) [inline jazzygrid/patterns/header.php]:
```suggestion
<p class="has-small-font-size"><a href="#" target="_blank" rel="noreferrer noopener"><?php echo __( 'Website', 'jazzygrid' ); ?></a></p>
<!-- /wp:paragraph -->

<!-- wp:paragraph {"fontSize":"small"} -->
<p class="has-small-font-size"><a href="#" target="_blank" rel="noreferrer noopener"><?php echo __( 'Albums', 'jazzygrid' ); ?></a></p>
<!-- /wp:paragraph -->

<!-- wp:paragraph {"fontSize":"small"} -->
<p class="has-small-font-size"><a href="#" target="_blank" rel="noreferrer noopener"><?php echo __( 'Tickets', 'jazzygrid' ); ?></a></p>
```

Should pattern text be translated? Also I assume 'Almuns' should be 'Albums'.

File: jazzygrid/patterns/header.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+<!-- /wp:cover -->
+
+<!-- wp:group {"align":"full","style":{"spacing":{"padding":{"top":"2rem","right":"2rem","bottom":"2rem","left":"2rem"},"margin":{"top":"0px","bottom":"0px"},"blockGap":"1.5rem"},"dimensions":{"minHeight":"100%"},"elements":{"link":{"color":{"text":"var:preset|color|background"}}}},"backgroundColor":"tertiary","textColor":"background","layout":{"type":"flex","orientation":"vertical","verticalAlignment":"bottom"}} -->
+<div class="wp-block-group alignfull has-background-color has-tertiary-background-color has-text-color has-background has-link-color" style="min-height:100%;margin-top:0px;margin-bottom:0px;padding-top:2rem;padding-right:2rem;padding-bottom:2rem;padding-left:2rem"><!-- wp:paragraph -->
+<p>The Jazz Messengers is a six-piece jazz ensemble known for their hard-bop style and virtuosic improvisation.</p>
```
Comment (ajlende) [inline jazzygrid/patterns/header.php]:
```suggestion
<p><?php echo esc_html__( 'The Jazz Messengers is a six-piece jazz ensemble known for their hard-bop style and virtuosic improvisation.', 'jazzygrid' ); ?></p>
```

File: jazzygrid/patterns/home.php [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+<!-- /wp:column -->
+
+<!-- wp:column {"verticalAlignment":"stretch","width":"25%","backgroundColor":"foreground"} -->
+<div class="wp-block-column is-vertically-aligned-stretch has-foreground-background-color has-background" style="flex-basis:25%"><!-- wp:group {"align":"full","style":{"spacing":{"padding":{"top":"1.5rem","right":"3rem","bottom":"1.5rem","left":"1.5rem"},"margin":{"top":"0px","bottom":"0px"},"blockGap":"1.5rem"},"dimensions":{"minHeight":"100%"},"elements":{"link":{"color":{"text":"var:preset|color|background"}}}},"textColor":"background","layout":{"type":"flex","orientation":"vertical","verticalAlignment":"top"}} -->
+<div class="wp-block-group alignfull has-background-color has-text-color has-link-color" style="min-height:100%;margin-top:0px;margin-bottom:0px;padding-top:1.5rem;padding-right:3rem;padding-bottom:1.5rem;padding-left:1.5rem"><!-- wp:heading -->
+<h2 class="wp-block-heading"><?php echo __('<a href="#" target="_blank" rel="noreferrer noopener">Newsletter</a>', 'jazzygrid');?></h2>
+<!-- /wp:heading -->
+
+<!-- wp:paragraph -->
+<p><?php echo __('Receive the latest updates about tickets &amp; events →', 'jazzygrid');?></p>
```
Comment (ajlende) [inline jazzygrid/patterns/home.php]: This feels like it should be a link.

File: jazzygrid/theme.json [hunk 4]
```diff
@@ -0,0 +1,713 @@
...
+			"area": "header",
+			"name": "header"
+		},
+		{
+			"area": "footer",
+			"name": "footer"
+		}
+	],
+	"version": 3,
+	"$schema": "https://schemas.wp.org/trunk/theme.json"
```
Comment (ajlende) [inline jazzygrid/theme.json:713]: Should match the minimum WP version to make sure all features are actually available. Same goes for style variations.

```suggestion
	"$schema": "https://schemas.wp.org/wp/6.0/theme.json"
```

There are a lot of things here that aren't available to WP 6.0 that need to get changed, or the minimum version needs to be updated.

File: jazzygrid/patterns/footer.php [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+<div class="wp-block-cover alignwide is-light has-custom-content-position is-position-top-left" style="margin-top:0px;margin-bottom:0px;padding-top:0px;padding-right:0px;padding-bottom:0px;padding-left:0px;min-height:10rem"><span aria-hidden="true" class="wp-block-cover__background has-custom-cce-7-e-2-background-color has-background-dim-0 has-background-dim"></span><div class="wp-block-cover__inner-container"><!-- wp:group {"align":"full","style":{"spacing":{"blockGap":"1rem"}},"layout":{"type":"default"}} -->
+<div class="wp-block-group alignfull"><!-- wp:heading -->
+<h2 class="wp-block-heading"><a href="https://en.wikipedia.org/wiki/Billie_Holiday" target="_blank" rel="noreferrer noopener">Innovative singer and composer</a></h2>
```
Comment (ajlende) [inline jazzygrid/patterns/footer.php]:
```suggestion
<h2 class="wp-block-heading"><?php echo __( '<a href="https://en.wikipedia.org/wiki/Billie_Holiday" target="_blank" rel="noreferrer noopener">Innovative singer and composer</a>', 'jazzygrid' ); ?></h2>
```

File: jazzygrid/patterns/footer.php [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+<!-- /wp:heading -->
+
+<!-- wp:paragraph -->
+<p>Along with interpretations of jazz standards, Holiday wrote many of her own songs — a rare feat for jazz singers of that time.</p>
```
Comment (ajlende) [inline jazzygrid/patterns/footer.php]:
```suggestion
<p><?php echo __( 'Along with interpretations of jazz standards, Holiday wrote many of her own songs — a rare feat for jazz singers of that time.', 'jazzygrid' ); ?></p>
```

File: jazzygrid/patterns/footer.php [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+<div class="wp-block-cover alignwide is-light has-custom-content-position is-position-top-left" style="margin-top:0px;margin-bottom:0px;padding-top:0px;padding-right:0px;padding-bottom:0px;padding-left:0px;min-height:10rem"><span aria-hidden="true" class="wp-block-cover__background has-custom-cce-7-e-2-background-color has-background-dim-0 has-background-dim"></span><div class="wp-block-cover__inner-container"><!-- wp:group {"align":"full","style":{"spacing":{"blockGap":"1rem"}},"layout":{"type":"default"}} -->
+<div class="wp-block-group alignfull"><!-- wp:heading -->
+<h2 class="wp-block-heading"><a href="https://www.pbs.org/wnet/americanmasters/billie-holiday-about-the-singer/68/" target="_blank" rel="noreferrer noopener">The Long Night of Lady Day</a></h2>
```
Comment (ajlende) [inline jazzygrid/patterns/footer.php]:
```suggestion
<h2 class="wp-block-heading"><?php echo __( '<a href="https://www.pbs.org/wnet/americanmasters/billie-holiday-about-the-singer/68/" target="_blank" rel="noreferrer noopener">The Long Night of Lady Day</a>', 'jazzygrid' ); ?></h2>
```

File: jazzygrid/patterns/footer.php [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+<!-- /wp:heading -->
+
+<!-- wp:group {"style":{"spacing":{"blockGap":"0.5rem"}},"layout":{"type":"flex","flexWrap":"nowrap"}} -->
+<div class="wp-block-group"><!-- wp:paragraph -->
+<p>Video</p>
```
Comment (ajlende) [inline jazzygrid/patterns/footer.php]:
```suggestion
<p><?php echo __( 'Video', 'jazzygrid' ); ?></p>
```

File: jazzygrid/patterns/home.php [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+<div class="wp-block-cover alignfull has-custom-content-position is-position-bottom-left" style="margin-top:0px;margin-bottom:0px;padding-top:1.5rem;padding-right:3rem;padding-bottom:1.5rem;padding-left:1.5rem;min-height:100%"><span aria-hidden="true" class="wp-block-cover__background has-background-dim-0 has-background-dim"></span><img class="wp-block-cover__image-background" alt="<?php echo __('', 'jazzygrid');?>" src="<?php echo esc_url( get_stylesheet_directory_uri() ); ?>/assets/images/beach-sea-coast-water-ocean-horizon.jpg" data-object-fit="cover"/><div class="wp-block-cover__inner-container"><!-- wp:site-logo {"width":120,"shouldSyncIcon":false,"align":"center"} /--></div></div>
```
Comment (ajlende) [inline jazzygrid/patterns/home.php]: I don't think empty strings need to be translated. This looks as if it's intended to be a decorative image, so the empty alt should be fine.

```suggestion
<div class="wp-block-cover alignfull has-custom-content-position is-position-bottom-left" style="margin-top:0px;margin-bottom:0px;padding-top:1.5rem;padding-right:3rem;padding-bottom:1.5rem;padding-left:1.5rem;min-height:100%"><span aria-hidden="true" class="wp-block-cover__background has-background-dim-0 has-background-dim"></span><img class="wp-block-cover__image-background" alt="" src="<?php echo esc_url( get_stylesheet_directory_uri() ); ?>/assets/images/beach-sea-coast-water-ocean-horizon.jpg" data-object-fit="cover"/><div class="wp-block-cover__inner-container"><!-- wp:site-logo {"width":120,"shouldSyncIcon":false,"align":"center"} /--></div></div>
```

File: jazzygrid/patterns/home.php [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+<p><?php echo __('The Jazz Messengers is a six-piece jazz ensemble known for their hard-bop style and virtuosic improvisation.', 'jazzygrid');?></p>
+<!-- /wp:paragraph -->
+
+<!-- wp:group {"style":{"spacing":{"blockGap":"1.5rem"}},"layout":{"type":"flex","flexWrap":"nowrap"}} -->
+<div class="wp-block-group"><!-- wp:paragraph {"fontSize":"small"} -->
+<p class="has-small-font-size"><?php echo __('<a href="#" target="_blank" rel="noreferrer noopener">Website</a>', 'jazzygrid');?></p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph {"fontSize":"small"} -->
+<p class="has-small-font-size"><?php echo __('<a href="#" target="_blank" rel="noreferrer noopener">Albuns</a>', 'jazzygrid');?></p>
```
Comment (ajlende) [inline jazzygrid/patterns/home.php]:
```suggestion
<p class="has-small-font-size"><?php echo __('<a href="#" target="_blank" rel="noreferrer noopener">Albums</a>', 'jazzygrid');?></p>
```

File: jazzygrid/patterns/home.php [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+<div class="wp-block-cover alignfull has-custom-content-position is-position-bottom-left" style="margin-top:0px;margin-bottom:0px;padding-top:1.5rem;padding-right:3rem;padding-bottom:1.5rem;padding-left:1.5rem;min-height:100%"><span aria-hidden="true" class="wp-block-cover__background has-background-dim-0 has-background-dim" style="background-color:#505050"></span><img class="wp-block-cover__image-background" alt="<?php echo __('', 'jazzygrid');?>" src="<?php echo esc_url( get_stylesheet_directory_uri() ); ?>/assets/images/img_trumpet-detail.png" data-object-fit="cover"/><div class="wp-block-cover__inner-container"><!-- wp:group {"style":{"spacing":{"padding":{"top":"0px","right":"0px","bottom":"0px","left":"0px"},"margin":{"top":"0px","bottom":"0px"},"blockGap":"0rem"}},"layout":{"type":"constrained"}} -->
```
Comment (ajlende) [inline jazzygrid/patterns/home.php]: Again, no need to translate an empty string of alt text for a decorative image.

```suggestion
<div class="wp-block-cover alignfull has-custom-content-position is-position-bottom-left" style="margin-top:0px;margin-bottom:0px;padding-top:1.5rem;padding-right:3rem;padding-bottom:1.5rem;padding-left:1.5rem;min-height:100%"><span aria-hidden="true" class="wp-block-cover__background has-background-dim-0 has-background-dim" style="background-color:#505050"></span><img class="wp-block-cover__image-background" alt="" src="<?php echo esc_url( get_stylesheet_directory_uri() ); ?>/assets/images/img_trumpet-detail.png" data-object-fit="cover"/><div class="wp-block-cover__inner-container"><!-- wp:group {"style":{"spacing":{"padding":{"top":"0px","right":"0px","bottom":"0px","left":"0px"},"margin":{"top":"0px","bottom":"0px"},"blockGap":"0rem"}},"layout":{"type":"constrained"}} -->
```

File: jazzygrid/patterns/home.php [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+<div class="wp-block-column is-vertically-aligned-stretch has-primary-background-color has-background" style="flex-basis:75%"><!-- wp:cover {"url":"<?php echo esc_url( get_stylesheet_directory_uri() ); ?>/assets/images/cover_trumpeteer.png","dimRatio":0,"isUserOverlayColor":true,"focalPoint":{"x":0.5,"y":0.40000000000000002},"className":"has-parallax is-repeated has-custom-content-position is-position-top-left"} -->
+<div class="wp-block-cover has-parallax is-repeated has-custom-content-position is-position-top-left"><span aria-hidden="true" class="wp-block-cover__background has-background-dim-0 has-background-dim"></span><img class="wp-block-cover__image-background " alt="<?php echo __('', 'jazzygrid');?>" src="<?php echo esc_url( get_stylesheet_directory_uri() ); ?>/assets/images/cover_trumpeteer.png" style="object-position:50% 40%" data-object-fit="cover" data-object-position="50% 40%"/><div class="wp-block-cover__inner-container"><!-- wp:group {"style":{"spacing":{"blockGap":"2rem","padding":{"right":"10%"}}},"layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch"}} -->
```
Comment (ajlende) [inline jazzygrid/patterns/home.php]:
```suggestion
<div class="wp-block-cover has-parallax is-repeated has-custom-content-position is-position-top-left"><span aria-hidden="true" class="wp-block-cover__background has-background-dim-0 has-background-dim"></span><img class="wp-block-cover__image-background " alt="" src="<?php echo esc_url( get_stylesheet_directory_uri() ); ?>/assets/images/cover_trumpeteer.png" style="object-position:50% 40%" data-object-fit="cover" data-object-position="50% 40%"/><div class="wp-block-cover__inner-container"><!-- wp:group {"style":{"spacing":{"blockGap":"2rem","padding":{"right":"10%"}}},"layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch"}} -->
```

File: jazzygrid/patterns/index.php
Comment (ajlende) [inline jazzygrid/patterns/index.php:1]: This whole file needs translations for strings.

**Review by henriqueiamarino — COMMENTED** [review]
Reviews are okay to me; thanks.

**Comment by henriqueiamarino** [comment]
Thanks, @ajlende. Since everything here is created with the CreateBlockTheme plugin, I'll see what I can do to fix the translation strings and the missing link.

File: jazzygrid/patterns/home.php [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+<div class="wp-block-group" style="min-height:100%;margin-top:0rem;margin-bottom:0rem;padding-top:4rem;padding-right:3rem;padding-bottom:4rem;padding-left:1.5rem"><!-- wp:group {"style":{"spacing":{"padding":{"top":"0px","right":"0px","bottom":"0px","left":"0px"},"margin":{"top":"0px","bottom":"0px"},"blockGap":"1.5rem"},"elements":{"link":{"color":{"text":"var:preset|color|foreground"}}}},"textColor":"foreground","layout":{"type":"flex","orientation":"vertical"}} -->
+<div class="wp-block-group has-foreground-color has-text-color has-link-color" style="margin-top:0px;margin-bottom:0px;padding-top:0px;padding-right:0px;padding-bottom:0px;padding-left:0px"><!-- wp:heading {"level":1} -->
+<h1 class="wp-block-heading"><?php echo __('<a href="https://en.wikipedia.org/wiki/Billie_Holiday" target="_blank" rel="noreferrer noopener"></a><a href="#" target="_blank" rel="noreferrer noopener">Jazzin\' Around Town</a>', 'jazzygrid');?></h1>
```
Comment (vcanales) [inline jazzygrid/patterns/home.php]: Seems like there's an empty `anchor` and then a used one, should these be just one anchor? Not sure if that's the correct URL for the text though

```suggestion
<h1 class="wp-block-heading"><?php echo __('<a href="https://en.wikipedia.org/wiki/Billie_Holiday" target="_blank" rel="noreferrer noopener">Jazzin\' Around Town</a>', 'jazzygrid');?></h1>
```

**Comment by henriqueiamarino** [comment]
@MaggieCabrera, here's the demo content file.

[jazzygrid.WordPress.2024-09-12.xml.zip](https://github.com/user-attachments/files/16982250/jazzygrid.WordPress.2024-09-12.xml.zip)

**Pull request merged**